services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/file/MultipartManagerV2Impl.java

```diff
@@ -114,35 +114,24 @@ <T extends MultipartRequest> MultipartUploadStatus startOrResumeMultipartRequest
 			request.setStorageLocationId(StorageLocationDAO.DEFAULT_STORAGE_LOCATION_ID);
 		}
 
-		// The MD5 is used to identify if this upload request already exists for this user.
-		String requestMD5Hex = MultipartRequestUtils.calculateMD5AsHex(request);
-
-		// Has an upload already been started for this user and request
-		CompositeMultipartUploadStatus status = multipartUploadDAO.getUploadStatus(user.getId(), requestMD5Hex);
-
-		boolean fileHandleExistsFileNoLongerInCloudStorage = fileHandleExistsFileNoLongerInCloudStorage(user, status);
-
-		if (forceRestart || fileHandleExistsFileNoLongerInCloudStorage) {
-			// When forcing a restart or if the underlying object key of a file handle does not exist we clear the old hash changing the cache key, the previous multipart upload will be garbage collected by a worker
-			multipartUploadDAO.setUploadStatusHash(user.getId(), requestMD5Hex, RESTARTED_HASH_PREFIX + requestMD5Hex + "_" + UUID.randomUUID().toString());
-		}
-
-		if (status == null || forceRestart || fileHandleExistsFileNoLongerInCloudStorage) {
+		CompositeMultipartUploadStatus status = getCachedUploadOrRestart(user, request, forceRestart).orElseGet(() -> {

```

Comment:
if (forceRestart) -> You can set the "cached upload" to an empty optional without even checking for a cached version.